views/js/payment-components/stored-payment-method.js

```diff
@@ -41,22 +40,32 @@ jQuery(function ($) {
 
     var threeDSProcessUrl;
 
-    $('[data-one-click-payment]').each(function (index, element) {
-        var oneClickPaymentConfiguration = $(element).data();
-        var oneClickPaymentMethod = oneClickPaymentConfiguration.oneClickPayment;
-        if (oneClickPaymentMethod) {
-            renderOneClickComponent(oneClickPaymentMethod);
-            fillBrowserInfo();
+    // use this object to iterate through the stored payment methods
+    var checkoutStoredPaymentMethods = window.adyenCheckout.paymentMethodsResponse.storedPaymentMethods;
+
+    // Iterate through the stored payment methods list we got from the adyen checkout component
+    checkoutStoredPaymentMethods.forEach(function (storedPaymentMethod) {
+
+        /*
+            storedPaymentMethod.id = $storedPaymentApiId in the stored-payment-method.tpl
+            if the container doesn't exits don't try to render the component
+        */
+        var storedPaymentMethodContainer = $("#cardContainer-" + storedPaymentMethod.id);
+
+        // container doesn't exist, something went wrong on the template side
+        if (!storedPaymentMethodContainer.length) {
+            return;
         }
 
-        threeDSProcessUrl = oneClickPaymentConfiguration.threeDsProcessUrl;
+        var storedPaymentMethodConfiguration = $('[data-stored-payment-api-id="' + storedPaymentMethod.id + '"]').data();
```

Comment:
can't we better do storedPaymentMethod.data() ?

Reply:
I checked with Attila as well. The elements are different and not in the same scope, so that's not possible without changing the HTML. I agree with your comment, though.